packages/@react-spectrum/s2/src/Table.tsx

```diff
@@ -11,85 +11,1103 @@
+
+  return (
+    <div style={{height: isResizing ? state?.tableHeight : '100%'}} className={resizerHandle({isFocusVisible, isHovered: isHovered || isInResizeMode, isResizing})} />
+  );
+}
+
+const tableHeader = style({
+  height: 'full',
+  width: 'full',
+  backgroundColor: 'gray-75'
+});
+
+const selectAllCheckbox = style({
+  marginStart: 16, // table-edge-to-content, same between mobile and desktop
+  marginEnd: {
+    default: 8,
+    scale: {
+      large: 14
+    }
+  }
+});
+
+const selectAllCheckboxColumn = style({
+  padding: 0,
+  height: '[calc(100% - 1px)]',
+  outlineStyle: 'none',
+  position: 'relative',
+  alignContent: 'center',
+  borderColor: 'gray-300',
+  borderXWidth: 0,
+  borderTopWidth: 0,
+  borderBottomWidth: 1,
+  borderStyle: 'solid',
+  backgroundColor: 'gray-75'
+});
+
+let InternalTableHeaderContext = createContext<{isHeaderRowHovered?: boolean}>({isHeaderRowHovered: false});
+
+export function TableHeader<T extends object>({columns, children}: TableHeaderProps<T>) {
+  let {scale} = useContext(InternalTableContext);
+  let {selectionBehavior, selectionMode} = useTableOptions();
+  let [isHeaderRowHovered, setHeaderRowHovered] = useState(false);
+
+  return (
+    <InternalTableHeaderContext.Provider value={{isHeaderRowHovered}}>
+      <RACTableHeader onHoverChange={setHeaderRowHovered} className={tableHeader}>
+        {/* Add extra columns for selection. */}
+        {selectionBehavior === 'toggle' && (
+          // Also isSticky prop is applied just for the layout, will decide what the RAC api should be later
+          // @ts-ignore
+          <RACColumn isSticky width={scale === 'medium' ? 40 : 48} minWidth={40} className={selectAllCheckboxColumn}>
+            {({isFocusVisible}) => (
+              <>
+                {selectionMode === 'single' &&
+                  <CellFocusRing isFocusVisible={isFocusVisible} />
+                }
+                {selectionMode === 'multiple' &&
+                  <Checkbox isEmphasized styles={selectAllCheckbox({scale})} slot="selection" />
+                }
+              </>
+            )}
+          </RACColumn>
+        )}
+        <Collection items={columns}>
+          {children}
+        </Collection>
+      </RACTableHeader>
+    </InternalTableHeaderContext.Provider>
+  );
+}
+
+const commonCellStyles = {
+  borderColor: 'transparent',
+  borderBottomWidth: 1,
+  borderTopWidth: 0,
+  borderXWidth: 0,
+  borderStyle: 'solid',
+  position: 'relative',
+  color: {
+    default: 'gray-800',
+    forcedColors: 'ButtonText'
+  },
+  outlineStyle: 'none',
+  paddingX: 16 // table-edge-to-content
+} as const;
+
+const cell = style<CellRenderProps & S2TableProps>({
+  ...commonCellStyles,
+  color: {
+    default: 'gray-800', // neutral-content-color-default
+    isHovered: 'gray-900', // neutral-content-color-hover
+    isPressed: 'gray-900', // neutral-content-color-down
+    isFocusVisible: 'gray-900' // neutral-content-color-key-focus
+  },
```

Comment:
Can't you use `color: "neutral"` here? That should set all of these states for you

Reply:
ah good call thanks, I'll do the same for the cell styles too.